developer-docs-site/docs/guides/nfts/mint-nft-cli.md

```diff
@@ -4,587 +4,291 @@ title: "Mint NFTs with the Aptos CLI"
 
-```shell
-cd aptos-move/move-examples/mint_nft/3-Adding-Admin
+```rust
+    const COLLECTION_DESCRIPTION: vector<u8> = b"Your collection description here!";
+    const TOKEN_DESCRIPTION: vector<u8> = b"Your token description here!";
+    // ...other const values
+    const TOKENS_BURNABLE_BY_CREATOR: bool = false;
+    const TOKENS_FREEZABLE_BY_CREATOR: bool = false;
 ```
 
-2. Edit `Move.toml` in that directory to replace `admin_addr = "0xcafe"` with the new `admin` address from `.aptos/config.yaml`.
+You can make these configurable on your own by adding them to the list of parameters sent into `initialize_collection` function below.
 
-3. Run the following CLI command to publish the module under a resource account, updating the `<seed>` and `default-account-address>` with your own:
+### Writing a collection creation function
 
-```shell
-aptos move create-resource-account-and-publish-package --seed <seed> --address-name mint_nft --profile default --named-addresses source_addr=<default-account-address>
+```rust
+public entry fun initialize_collection(
+    creator: &signer
+    collection_name: String,
+    collection_uri: String,
+    maximum_supply: u64,
+    royalty_numerator: u64,
+    royalty_denominator: u64,
+    token_name: String,
+    token_uri: String,
+) acquires MintConfiguration {
+    // ensure the signer of this function call is also the owner of the contract
+    let creator_addr = signer::address_of(creator);
+    assert!(creator_addr == @mint_nft, error::permission_denied(ENOT_AUTHORIZED));
+
+    aptos_token::create_collection(
+        creator,
+        string::utf8(COLLECTION_DESCRIPTION),
+        maximum_supply,
+        collection_name,
+        collection_uri,
+        MUTABLE_COLLECTION_DESCRIPTION,
+        MUTABLE_ROYALTY,
+        MUTABLE_URI,
+        MUTABLE_TOKEN_DESCRIPTION,
+        MUTABLE_TOKEN_NAME,
+        MUTABLE_TOKEN_PROPERTIES,
+        MUTABLE_TOKEN_URI,
+        TOKENS_BURNABLE_BY_CREATOR,
+        TOKENS_FREEZABLE_BY_CREATOR,
+        royalty_numerator,
+        royalty_denominator,
+    );
+    move_to(creator, MintConfiguration {
+        collection_name,
+        token_name,
+        token_uri,
+    });
+}
 ```
 
-4. As usual, receive output asking you to accept a gas fee for the transaction and enter: `yes`.
+First off, note the first couple lines of the function:
 
-5. Receive the output:
-
-```shell
-"Result": "Success"`
+```rust
+let creator_addr = signer::address_of(creator);
+assert!(creator_addr == @mint_nft, error::permission_denied(ENOT_AUTHORIZED));
 ```
 
-### Include admin functions
+This gates the ability to run this function to the owner of the contract. This isn't necessary, but unless you specifically design a contract around
+letting multiple accounts use it, it's simpler to only allow the account that deploys the contract to use it. This avoids a few potential vulnerabilities where you unintentionally allow an account to modify internal module resources for another account.
```

Comment:
Let's mention we'll show to migrate from deployer-only to an admin model in later steps (parts?)